src/components/partial-string-match/PartialStringMatch.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+export interface PartialStringMatchProps {
+    wholeString: string;
+    partialMatch?: string;
+    caseInsensitive?: boolean;
+}
+
+export const PartialStringMatch = (props: PartialStringMatchProps): JSX.Element => {
+    const wholeString = props.wholeString || '';
+    const partialMatch = props.partialMatch || '';
+    const regExp = new RegExp(partialMatch, props.caseInsensitive && 'i');
+    const splitPlaceholder = '%%split%%';
+
+    let hasMatch = false;
+    let realMatch: string;
+    let finalString: string | string[] = wholeString.replace(regExp, (match: string) => {

```

Comment:
You could name it something else instead of using a let and casting to both string and array of string. Only the one on line 27 is used anyway,

Reply:
both are used but I'll refactor as suggested 👍